py4cast/observer.py

```diff
@@ -422,7 +376,7 @@ def update(
                 spatial_loss, "b t (x y) -> b t x y ", x=obj.grid_shape[0]
             )
         self.spatial_loss_maps.append(
-            reduce(obj, spatial_loss).unsqueeze(0)
+            obj.trainer.strategy.reduce(spatial_loss, reduce_op="mean").cpu()
```

Comment:
What about the `unsqueeze` op  that has been erased ? is it voluntary ?

Reply:
Yes it is voluntary.

Well there was a flatten(0, 1) in one condition (Distributed) of the former reduce function followed by this unsqueeze(0). Made no sense. 

In SingleDevice there was no flatten(0, 1) and there was an exception raised in the plot code later on because there was an extra dimension in the tensors.